File: Codes/2210-count-hills-and-valleys-in-an-array.py
'''
Author: LetMeFly
Date: 2025-07-28 21:39:52
LastEditors: LetMeFly.xyz
LastEditTime: 2025-07-28 22:20:34
'''
from typing import List

class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        last = nums[0]
        ans = 0
        for i, t in enumerate(nums[1:-1]):
            ans += t > last and t > nums[i + 1] or t < last and t < nums[i + 1]
            if nums[i] != nums[i + 1]:
                last = nums[i]
        return ans

File: Codes/2210-count-hills-and-valleys-in-an-array.java
/*
 * @Author: LetMeFly
 * @Date: 2025-07-28 21:39:52
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-07-28 22:25:46
 */
class Solution {
    public int countHillValley(int[] nums) {
        int last = nums[0];
        int ans = 0;
        for (int i = 1; i < nums.length - 1; i++) {
            if (nums[i] > last && nums[i] > nums[i + 1] || nums[i] < last && nums[i] < nums[i + 1]) {
                ans++;
            }
            if (nums[i] != nums[i + 1]) {
                last = nums[i];
            }
        }
        return ans;
    }
}